drivers/spi/Kconfig.max32

```diff
@@ -23,4 +23,22 @@ config SPI_MAX32_DMA
 	help
 	  Enable DMA support for MAX32 MCU SPI driver.
 
+if SPI_RTIO
```

Comment:
Shouldn't we rely on CONFIG_SPI_MAX32_INTERRUPT?

Reply:
Sorry if I wasn't clear enough: I think we should have this driver select `SPI_MAX32_INTERRUPT`, as that seems to be the way it would work in a non-blocking fashion.

RTIO implementors should be non-blocking/asynchronous to the caller by definition.